spring-integration-core/src/main/java/org/springframework/integration/support/converter/MapMessageConverter.java

```diff
@@ -43,10 +43,12 @@
  */
 public class MapMessageConverter implements MessageConverter, BeanFactoryAware {
 
-	private volatile String[] headerNames;
+	// TODO is initializing with an empty array correct or should toMessage() return 'null' if never set?
+	private volatile String[] headerNames = {};
```

Comment:
This is probably not desired. Should I add something like `Assert.notEmpty(this.headerNames, "at least one header name must be set with setHeaderNames()");` for `toMessage()` and `fromMessage()`

Reply:
I think we need to make it nullable and implement respective logic.
It is going to be more efficient to avoid some extra calls if this one is just null.

Reply:
The question still remains, what logic should be implemented. Should `this.headerNames == null` be just handled like an empty array or is a different behavior needed?

Reply:
I think what you did is enough.
It really looked like a bug by default where this `headerNames` is null and it is not handled properly.
Now it is OK. 
Let's remove that TODO!
Thank you!

Reply:
Fine. I adapted `setHeaderNames(String... headerNames)` to use an empty array if called with `null` or an empty array